npm/private/lifecycle/lifecycle-hooks.js

```diff
@@ -68,6 +69,46 @@ async function makeBins(nodeModulesPath, scope, segmentsUp) {
     }
 }
 
+// Helper from not exported by @pnpm/lifecycle
+async function checkBindingGyp(root, scripts) {
+    if (await exists(path.join(root, 'binding.gyp'))) {
+        scripts['install'] = 'node-gyp rebuild'
+    }
+}
+
+// Like runPostinstallHooks from @pnpm/lifecycle but also runs prepare hook which is notably missing
+// from that function and does not match what pnpm does on its install:
+// https://github.com/pnpm/pnpm/blob/bc18d33fe00d9ed43f1562d4cc6d37f49d9c2c38/pkg-manager/core/src/install/index.ts#L1108
```

Comment:
This seems to run them concurrently? The method also seems to be public from `@pnpm/lifecycle`, do/can we depend on that?

Reply:
The concurrently method is a lot more than I'm comfortable changing in this PR. Want to keep the scope limited to just adding the `prepare` hook on top of what we currently have. Could be explored in a follow-up at some point.